zhaquirks/danfoss/thermostat.py

```diff
@@ -26,113 +55,456 @@
+
+    class AttributeDefs(Diagnostic.AttributeDefs):
+        """Attribute Definitions."""
+
+        sw_error_code = ZCLAttributeDef(
+            id=0x4000,
+            type=DanfossSoftwareErrorCodeBitmap,
+            access="rpw",
+            is_manufacturer_specific=True,
+        )
+        wake_time_avg = ZCLAttributeDef(
+            id=0x4001, type=types.uint32_t, access="r", is_manufacturer_specific=True
+        )
+        wake_time_max_duration = ZCLAttributeDef(
+            id=0x4002, type=types.uint32_t, access="r", is_manufacturer_specific=True
+        )
+        wake_time_min_duration = ZCLAttributeDef(
+            id=0x4003, type=types.uint32_t, access="r", is_manufacturer_specific=True
+        )
+        sleep_postponed_count_avg = ZCLAttributeDef(
+            id=0x4004, type=types.uint32_t, access="r", is_manufacturer_specific=True
+        )
+        sleep_postponed_count_max = ZCLAttributeDef(
+            id=0x4005, type=types.uint32_t, access="r", is_manufacturer_specific=True
+        )
+        sleep_postponed_count_min = ZCLAttributeDef(
+            id=0x4006, type=types.uint32_t, access="r", is_manufacturer_specific=True
+        )
+        motor_step_counter = ZCLAttributeDef(
+            id=0x4010, type=types.uint32_t, access="rp", is_manufacturer_specific=True
+        )
+
+        data_logger = ZCLAttributeDef(
+            id=0x4020,
+            type=types.LimitedLVBytes(50),
+            access="rpw",
+            is_manufacturer_specific=True,
+        )
+        control_diagnostics = ZCLAttributeDef(
+            id=0x4021,
+            type=types.LimitedLVBytes(30),
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+        control_diagnostics_frequency = ZCLAttributeDef(
+            id=0x4022, type=types.uint16_t, access="rw", is_manufacturer_specific=True
+        )  # non-configurable reporting
+
+
+class DanfossTimeCluster(CustomizedStandardCluster, Time):
+    """Danfoss cluster for fixing the time."""
+
+    async def write_time(self):
+        """Write time info to Time Cluster.
+
+        It supports adjusting for daylight saving time, but this is not trivial to retrieve with the modules:
+            zoneinfo, datetime or time
+        """
+        epoch = datetime(2000, 1, 1, 0, 0, 0, 0, tzinfo=UTC)
+        current_time = (datetime.now(UTC) - epoch).total_seconds()
+
+        await self.write_attributes(
+            {
+                "time": current_time,
+                "time_status": 0b00000010,  # only bit 1 can be set
+                "time_zone": time.timezone
+            }
+        )
+
+    async def bind(self):
+        """According to the documentation of Zigbee2MQTT there is a bug in the Danfoss firmware with the time.
+
+        It doesn't request it, so it has to be fed the correct time.
+        """
+        result = await super().bind()
+
+        await self.write_time()
+
+        return result
```

Comment:
Last thing, we can remove these two empty lines:
```suggestion
        result = await super().bind()
        await self.write_time()
        return result
```